File: src/api/rest_endpoints/health.rs
//! Health check endpoints for the REST API

use axum::response::IntoResponse;
use axum::{http::StatusCode, Json};
use chrono::Utc;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
struct HealthResponse {
    status: String,
    timestamp: String,
    service: String,
}

/// Health check endpoint
pub async fn health_check() -> impl IntoResponse {
    let response = HealthResponse {
        status: "healthy".to_string(),
        timestamp: Utc::now().to_rfc3339(),
        service: "uveddi-api".to_string(),
    };
    (StatusCode::OK, Json(response))
}

File: api-server/src/routes/health.js
const express = require('express');
const router = express.Router();

// Health check endpoint
router.get('/health', (req, res) => {
  res.json({
    status: 'healthy',
    timestamp: new Date().toISOString(),
    version: '0.9.0-alpha',
    service: 'uveddi-api-server'
  });
});

// V1 health check
router.get('/api/v1/health', (req, res) => {
  res.json({
    status: 'ok',
    timestamp: new Date().toISOString(),
    service: 'Uveddi API',
    version: 'v1.0'
  });
});

// V1 metrics endpoint
router.get('/api/v1/metrics', (req, res) => {
  res.json({
    uptime: process.uptime(),
    memory: process.memoryUsage(),
    timestamp: new Date().toISOString()
  });
});

module.exports = router;